expression/collation.go

```diff
@@ -189,12 +168,102 @@ func deriveCoercibilityForColumn(c *Column) Coercibility {
 	return CoercibilityImplicit
 }
 
+func deriveCollation(ctx sessionctx.Context, funcName string, args []Expression, retType types.EvalType, argTps ...types.EvalType) (dstCharset, dstCollation string, coercibility Coercibility, err error) {
+	switch funcName {
```

Comment:
Please write a UT for deriveCollation(), we need more tests for it.

Reply:
will add more UT in the next PR because the next PR will make the aggregate the same with MySQL.